File: tmp_six_lang_depth_repo/go/service.go
package service

import "errors"

type Runner interface {
	Run(value string) string
}

type UserService struct {
	count int
}

func (s *UserService) Run(value string) string {
	return value
}

func (s *UserService) Validate(value string) (string, error) {
	s.count = s.count + 1
	if value == "" {
		return "", errors.New("missing")
	}
	return s.Run(value), nil
}

func Entry(value string) (string, error) {
	service := &UserService{}
	return service.Validate(value)
}

File: tmp_six_lang_depth_repo/java/Service.java
interface Runner {
    String run(String value);
}

class BaseError extends RuntimeException {
    BaseError(String message) {
        super(message);
    }
}

class ServiceError extends BaseError {
    ServiceError(String message) {
        super(message);
    }
}

class UserService implements Runner {
    private int count = 0;

    public String run(String value) {
        return value;
    }

    public String validate(String value) {
        this.count = this.count + 1;
        if (value == null) {
            throw new ServiceError("missing");
        }
        return this.run(value);
    }
}

class EntryPoint {
    static String entry(String value) {
        UserService service = new UserService();
        return service.validate(value);
    }
}
